### ISSUE TITLE
[no-array-method-this-argument] - false positives, lodash map, foreach, filter and React.Chidlren.map

### ISSUE THREAD

**Issue by ChrisWiles** [sindresorhus/eslint-plugin-unicorn#1385]
```js
          const files = _.filter(data, (file) => _.endsWith(file, '.pdf')); //  Do not use the `this` argument in `Array#filter()`  unicorn/no-array-method-this-argument
        
          {React.Children.map(children, (child) => // Do not use the `this` argument in `Array#map()`  unicorn/no-array-method-this-argument
            React.cloneElement(child, { className: classes.nested })
          )}
          
          
```

**Comment by jan-molak** [comment]
Also false positives with custom classes that happen to have a `map` method.

```typescript
class Mapper {
  map(argument) {
    // ...
  }
}
```

**Comment by fungiboletus** [comment]
Also with `Vue.filter`.

**Comment by fisker** [comment]
> Also false positives with custom classes that happen to have a `map` method.
> 
> ```ts
> class Mapper {
>   map(argument) {
>     // ...
>   } 
> }
> ```

You'll have to use `eslint-disable` for your case, other cases will be ignored https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1386

**Comment by hi-rustin** [comment]
Is it a false positive?

In emberjs:

https://github.com/rust-lang/crates.io/blob/396a79d4a5e7d23bb64282031c65438a956deb6a/app/routes/category.js#L10

**Comment by fisker** [comment]
Yes, it is.

**Comment by hi-rustin** [comment]
> Is it a false positive?
> 
> In emberjs:
> 
> https://github.com/rust-lang/crates.io/blob/396a79d4a5e7d23bb64282031c65438a956deb6a/app/routes/category.js#L10

This code still has false positives.

cc: @fisker  @sindresorhus

**Comment by fisker** [comment]
Can you update to `eslint-plugin-unicorn` to v34.0.1 ? https://github.com/rust-lang/crates.io/pull/3759

**Comment by hi-rustin** [comment]
> Can you update to `eslint-plugin-unicorn` to v34.0.1 ? [rust-lang/crates.io#3759](https://github.com/rust-lang/crates.io/pull/3759)

It's fixed. Thanks!
